azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -91,20 +121,74 @@ void restartExecutableFlow(final ExecutableFlow originalExFlow) {
       return;
     }
 
-    // update the original executable-flow with retry-count and child executionID
     try {
       originalExFlow.setFlowRetryChildExecutionID(retryExFlow.getExecutionId());
       this.executorLoader.updateExecutableFlow(originalExFlow);
     } catch (final ExecutorManagerException e) {
       logger.error("Failed to update the original flow after restart"
           + originalExFlow.getFlowId() + ". " + e.getMessage());
     }
+
     // TODO: consider send out email for this information
+    // update the original executable-flow with retry-count and child executionID
     logger.info(String.format("Retry execution [%d] successfully, "
             + "spawning child-execution [%d], and its root-execution was [%d];"
             + "system-defined retry count=%d, user-defined retry-count=%d.",
         originalExFlow.getExecutionId(), retryExFlow.getExecutionId(),
         retryExFlow.getFlowRetryRootExecutionID(),
         retryExFlow.getSystemDefinedRetryCount(), retryExFlow.getUserDefinedRetryCount()));
   }
+
+
+  /**
+   * DFS walk through of the executableFlow, update the retry-flow's job status to disabled if
+   * the corresponding job status in origin-flow is considered success/skip/disable.
+   */
+  public static void disableSucceededSkippedJobsInRetryFlow(
```

Comment:
I checked this route, and I believe its a big way around:
1. DFS the original execution to find out the succeeded nodes
2. put into initiallyDisabledJobs
3. at new execution start time, apply initiallyDisabledJobs with DFS

Also, with this PR, users CAN see the nodes are being disabled by my algorithm for the new execution from UI, and if they want to restart it with the same setting, it's totally possible.
This is because when webUI display the nodes' status, it's from `ExecutableNode.status`, and when it submit execution via http call, the `disabled job` list are constructed by frontend, but not from the DB-stored `ExecutionOptions`

Reply:
I see, thanks for thinking about this!
My concern is that the current code uses the original ExecutionOptions and overwrites attributes as needed and we're never modifying the list of disabled nodes so original configs will be set for retries. What nodes will ultimately be executed if the original execution had disabled nodes and the user specified retry strategy disable succeeded nodes?

Reply:
According to the current code, below explains what's the ultimate state of nodes:
1. original execution being submitted with `DisabledJobs` specified
2. original execution data-structure being applied with these `DisabledJobs`, the nodes got marked statues `Disabled`
3. original execution got failed/exec-stopped, gets into retry
4. If using disabled-succeeded-nodes strategy: DFS walk through both executions, disable retry-execution's nodes whose corresponding node is `disabled` or `succeeded` in original execution.
5. retry-execution applied with these `DisabledJobs`, but because 4) is done, this == no-op.

To sum up: all that's originally disabled in original-execution, and those succeeded in original-execution, will be disabled in retry-execution.